src/OpenSkill/Datatable/Providers/QueryBuilderProvider.php

```diff
@@ -0,0 +1,251 @@
+        $this->compileQuery($this->columnConfiguration);
+
+        // sort
+        $this->sortQuery();
+
+        // limit
+        $this->queryBeforeLimits = clone $this->query;
+        $this->limitQuery();
+
+        // original # of items
+        $filteredItems = $this->originalQuery->count();
+
+        // # of items in filtered & ordered dataset
+        $dataCount = $this->queryBeforeLimits->count();
+
+        // the data for the response
+        $columns = $this->compileColumnNames();
+        $response = new Collection($this->query->get($columns));
+
+        // slice the result into the right size
+        return new ResponseData(
+            $response,
+            $filteredItems,
+            $dataCount
+        );
+    }
+
+    /**
+     * Will compile the collection into the final collection where operations like search and order can be applied.
+     *
+     * @param QueryBuilder $query
+     * @param ColumnConfiguration[] $columnConfiguration
+     * @return QueryBuilder
+     * @throws DatatableException
+     */
+    private function compileQuery(array $columnConfiguration)
+    {
+        if ($this->queryConfiguration->isGlobalSearch()) {
+            $this->compileGlobalQuery($columnConfiguration);
+        } elseif ($this->queryConfiguration->isColumnSearch()) {
+            $this->compileColumnQuery($columnConfiguration);
+        }
+    }
+
+    /**
+     * When a global (single) search has been done against data in the datatable.
+     *
+     * @param array $columnConfiguration
+     * @return QueryBuilder
+     * @throws DatatableException
+     */
+    private function compileGlobalQuery(array $columnConfiguration)
+    {
+        foreach ($columnConfiguration as $i => $col) {
+            $this->createQueryForColumn($col, $this->queryConfiguration->searchValue());
+        }
+    }
+
+    /**
+     * When a global query is being performed (ie, a query against a single column)
+     *
+     * @param ColumnConfiguration[] $columnConfiguration
+     * @return QueryBuilder
+     * @throws DatatableException
+     */
+    private function compileColumnQuery(array $columnConfiguration)
+    {
+        $searchColumns = $this->queryConfiguration->searchColumns();
+
+        foreach ($searchColumns as $i => $col) {
+            $column = $this->getColumnFromName($columnConfiguration, $col->columnName());
+
+            if (!isset($column))
+                continue;
+
+            $this->createQueryForColumn($column, $col->searchValue());
+        }
+    }
+
+    private function createQueryForColumn(ColumnConfiguration $column, $searchValue)
+    {
+        $searchType = $column->getSearch();
+
+        if ($searchType == DefaultSearchable::NONE()) {
+            // Don't do anything, this is not a searchable field
+            return $this->query;
+        } elseif ($searchType == DefaultSearchable::NORMAL()) {
+            $this->query->orWhere($column->getName(), 'LIKE', '%' . $searchValue . '%');
```

Comment:
We already really have the option to override any column query with the Searchable types, so we just need to add one for case insensitive as another option.

The only thing is that then that the `QueryBuilderProvider` would need to know what database it is running on so that we can build the correct query depending on wether the database is MySQL or Postgres.

Maybe adding `->isPostgres` and `->isMySQL` as functions on this class might not be a bad idea? Then at least we can throw errors if we can't build the correct query for a db...

Either way, I am not really worried about this until someone opens a bug or feature request :)